src/Elastic.Apm.AspNetCore/Extensions/TransactionExtentions.cs

```diff
@@ -1,36 +1,52 @@
 using System.Net.Mime;
-using Elastic.Apm.Api;
 using Elastic.Apm.Config;
 using Elastic.Apm.Logging;
+using Elastic.Apm.Model;
 using Microsoft.AspNetCore.Http;
 
 namespace Elastic.Apm.AspNetCore.Extensions
 {
-	public static class TransactionExtensions
+	internal static class TransactionExtensions
 	{
 		/// <summary>
-		/// Collects the Request body (and possibly additional information in the future)
-		/// in the transaction
+		/// Collects the Request body in the transaction
 		/// </summary>
 		/// <param name="transaction">Transaction object</param>
-		/// <param name="httpContext">Current http context</param>
+		/// <param name="isForError">Is request body being captured for error (otherwise it's for transaction)</param>
+		/// <param name="request">Current http request</param>
 		/// <param name="logger">Logger object</param>
-		public static void CollectRequestInfo(this ITransaction transaction, HttpContext httpContext, IConfigurationReader configurationReader,
-			IApmLogger logger
-		)
+		internal static void CollectRequestBody(this Transaction transaction, bool isForError, HttpRequest request, IApmLogger logger)
 		{
-			var body = Consts.BodyRedacted; // According to the documentation - the default value of 'body' is '[Redacted]'
+			if (!transaction.IsSampled) return;
 
-			// We need to parse the content type and check it's not null and is of valid value 
-			if (!string.IsNullOrEmpty(httpContext?.Request?.ContentType))
-			{
-				var contentType = new ContentType(httpContext.Request.ContentType);
+			// Is request body already captured?
+			if (transaction.IsContextCreated
```

Comment:
Why does `transaction.IsContextCreated` matter here? 🤔 Is it because `transaction.Context.Request` would create one? If so, is there any case when only the `Body` is filled? I don't think so... so if we are here, I think we already have a context anyway, or not?

Reply:
I didn't want to assume too much about the context in which this method is called - the purpose was to avoid creating empty `Context`. I've added a comment to the code.

Reply:
Ok, sounds good to me.